internal/pkg/gateway/api.go

```diff
@@ -427,11 +428,81 @@ func (gs *Server) Submit(ctx context.Context, request *gp.SubmitRequest) (*gp.Su
 		return nil, status.Errorf(codes.Unavailable, "no orderer nodes available")
 	}
 
+	logger := logger.With("txID", request.TransactionId)
+	config := gs.getChannelConfig(request.ChannelId)
+	if config.ChannelConfig().Capabilities().ConsensusTypeBFT() {
+		return gs.submitBFT(ctx, orderers, txn, config, logger)
+	} else {
+		return gs.submitNonBFT(ctx, orderers, txn, logger)
+	}
+}
+
+func (gs *Server) submitBFT(ctx context.Context, orderers []*orderer, txn *common.Envelope, config channelconfig.Resources, logger *flogging.FabricLogger) (*gp.SubmitResponse, error) {
+	// For BFT, we send transaction to ALL orderers
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	waitCh := make(chan *gp.ErrorDetail, len(orderers))
+	for _, o := range orderers {
+		go func(ord *orderer) {
+			logger.Infow("Sending transaction to orderer", "endpoint", ord.logAddress)
+			ctx, cancel := context.WithTimeout(ctx, gs.options.BroadcastTimeout)
+			defer cancel()
+			response, err := gs.broadcast(ctx, ord, txn)
+
+			if err != nil {
+				logger.Warnw("Error sending transaction to orderer", "endpoint", ord.logAddress, "err", err)
+				waitCh <- errorDetail(ord.endpointConfig, err.Error())
+			} else if status := response.GetStatus(); status == common.Status_SUCCESS {
+				logger.Infow("Successful response from orderer", "endpoint", ord.logAddress)
+				waitCh <- nil
+			} else {
+				logger.Warnw("Unsuccessful response sending transaction to orderer", "endpoint", ord.logAddress, "status", status, "info", response.GetInfo())
+				waitCh <- errorDetail(ord.endpointConfig, fmt.Sprintf("received unsuccessful response from orderer: %s", common.Status_name[int32(status)]))
+			}
+		}(o)
+	}
+	n := len(orderers)
+	ordererConfig, ok := config.OrdererConfig()
+	if ok {
```

Comment:
what if not ok?

Reply:
If it's not ok, then it defaults to the number of (channel) orderers that the gateway is connected to (& cached in the registry).  This should be the same, since the registry gets the connection information from the same OrdererConfig.  However, if the gateway was unable to connect to an orderer, then these numbers could be different.  It depends whether we want to calculate the consensus from the number of connected orderers, or the number of orderers defined in the channel config.  Either way, we can only attempt to broadcast to the connected ones.